gen/struct_test.go

```diff
@@ -877,112 +877,131 @@ func TestStructWithDefaults(t *testing.T) {
 
 func TestStructJSON(t *testing.T) {
 	tests := []struct {
-		v interface{}
-		j string
+		v, w interface{}
+		j    string
 	}{
-		{&ts.Point{X: 0, Y: 0}, `{"x":0,"y":0}`},
-		{&ts.Point{X: 1, Y: 2}, `{"x":1,"y":2}`},
+		{v: &ts.Point{X: 0, Y: 0}, j: `{"x":0,"y":0}`},
+		{v: &ts.Point{X: 1, Y: 2}, j: `{"x":1,"y":2}`},
 		{
-			&ts.Edge{
+			v: &ts.Edge{
 				StartPoint: &ts.Point{X: 1, Y: 2},
 				EndPoint:   &ts.Point{X: 3, Y: 4},
 			},
-			`{"startPoint":{"x":1,"y":2},"endPoint":{"x":3,"y":4}}`,
+			j: `{"startPoint":{"x":1,"y":2},"endPoint":{"x":3,"y":4}}`,
 		},
 		{
-			&ts.Edge{StartPoint: &ts.Point{X: 1, Y: 1}},
-			`{"startPoint":{"x":1,"y":1},"endPoint":null}`,
+			v: &ts.Edge{StartPoint: &ts.Point{X: 1, Y: 1}},
+			j: `{"startPoint":{"x":1,"y":1},"endPoint":null}`,
 		},
 		{
-			&ts.Edge{
+			v: &ts.Edge{
 				StartPoint: &ts.Point{X: 1, Y: 1},
 				EndPoint:   &ts.Point{X: 1, Y: 1},
 			},
-			`{"startPoint":{"x":1,"y":1},"endPoint":{"x":1,"y":1}}`,
+			j: `{"startPoint":{"x":1,"y":1},"endPoint":{"x":1,"y":1}}`,
 		},
-		{&ts.User{Name: ""}, `{"name":""}`},
-		{&ts.User{Name: "foo"}, `{"name":"foo"}`},
+		{v: &ts.User{Name: ""}, j: `{"name":""}`},
+		{v: &ts.User{Name: "foo"}, j: `{"name":"foo"}`},
 		{
-			&ts.User{
+			v: &ts.User{
 				Name:    "foo",
 				Contact: &ts.ContactInfo{EmailAddress: "bar@example.com"},
 			},
-			`{"name":"foo","contact":{"emailAddress":"bar@example.com"}}`,
+			j: `{"name":"foo","contact":{"emailAddress":"bar@example.com"}}`,
 		},
 		{
-			&ts.User{
+			v: &ts.User{
 				Name:    "foo",
 				Contact: &ts.ContactInfo{EmailAddress: ""},
 			},
-			`{"name":"foo","contact":{"emailAddress":""}}`,
+			j: `{"name":"foo","contact":{"emailAddress":""}}`,
 		},
-		{&tu.EmptyUnion{}, "{}"},
-		{&tu.Document{Pdf: td.PDF("hello")}, `{"pdf":"aGVsbG8="}`},
-		{&tu.Document{Pdf: td.PDF{}}, `{"pdf":""}`},
+		{v: &tu.EmptyUnion{}, j: "{}"},
+		{v: &tu.Document{Pdf: td.PDF("hello")}, j: `{"pdf":"aGVsbG8="}`},
+		{v: &tu.Document{Pdf: td.PDF{}}, w: &tu.Document{Pdf: nil}, j: `{}`},
```

Comment:
Does the previously written test case continue to pass? Might be better to just replace, remove, or move.